### ISSUE TITLE
Support for Kotlin enum classes?

### ISSUE THREAD

**Issue by tudorgk** [evernote/android-state#8]
Hi again,

Java enums seem to work, but when trying to serialise kotlin enum classes, I get a stack overflow when processing annotations.

Do you have any knowledge of this issue, or I'm doing something wrong here?

Cheers!

**Comment by maxost** [comment]
(just walking around) I have no problems with Kotlin enum classes. Please provide some code.

**Comment by vRallev** [comment]
I can reproduce the issue fairly easy with following sample, thanks for pointing it out!
```kotlin
enum class Test {
    LEFT, RIGHT
}

class TestKotlin {
    @State
    var test = Test.LEFT
}
```

**Comment by vRallev** [comment]
Fixed in version `1.0.4`.
